WiseBuy/Managers/APIManager.m

```diff
@@ -24,16 +43,125 @@ + (void)fetchDealsFromEbayAPI {
     completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
       if (error) {
         NSLog(@"%@", error);
+        dispatch_semaphore_signal(sema);
       } else {
+//            NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *) response;
+            NSError *parseError = nil;
+            NSDictionary *responseDictionary = [NSJSONSerialization JSONObjectWithData:data options:0 error:&parseError];
+            NSLog(@"%@",responseDictionary);
+            dispatch_semaphore_signal(sema);
+            
+      }
+    }];
+    [dataTask resume];
+    dispatch_semaphore_wait(sema, DISPATCH_TIME_FOREVER);
+}
+
++ (void)fetchDealsFromUPCDatabase{
```

Comment:
(I thought I wrote a comment about this, but I don't see it now...re-writing it)

Having one method per API is fine to start, but as a suggestion for a future improvement, think about how to split up this code to make it more scalable when multiple APIs are added. Putting the APIs all into this one file would make this file gigantic, and is hard to maintain with multiple developers working at the same time.